lib/ui/results.js

```diff
@@ -0,0 +1,338 @@
+      };
+    }
+    return null;
+  },
+
+
+  _getContentMaxWidth: function(result) {
+    /*
+    This method is bad.
+
+    It attempts to calculate the available space for the recommendation content
+    (Title, URL) by subtracting a magic number from the width of the popup.
+
+    That magic number estimates the amount of horizontal space that non-content
+    elements are taking up. The calculation for that number:
+
+      9px recommendation left padding
+     16px favicon
+      8px favicon right margin
+     32px image/letterbox
+      8px image/letterbox right margin
+      8px label left margin
+    100px label width (appx)
+     15px recommendation right padding
+    -----
+    197px
+
+    If the available space is less than 200px, we will grow the popup to ensure
+    that there are at least 200px available for it.
+    */
+    const MIN = 200;
+    const MAGIC = 197;
+    const popupWidth = Math.floor(
+      result.doc.getElementById('PopupAutoCompleteRichResult').width);
+    const available = popupWidth - MAGIC;
+    return available >= MIN ? available : MIN;
+  }
+};
+
+
+
+function TLDResult(data, opts) {
+  BaseResult.call(this, data, opts);
+}
+TLDResult.prototype = Object.create(BaseResult.prototype);
+TLDResult.prototype.constructor = TLDResult;
+TLDResult.prototype.getType = function(result) {
+  return 'tld';
+};
+TLDResult.prototype.getImageUrl = function(result) {
+  return result._get('enhancements.tld');
+};
+
+
+function WikipediaResult(data, opts) {
+  BaseResult.call(this, data, opts);
+}
+WikipediaResult.prototype = Object.create(BaseResult.prototype);
+WikipediaResult.prototype.constructor = WikipediaResult;
+WikipediaResult.prototype.getType = function(result) {
+  return 'wikipedia';
+};
+WikipediaResult.prototype.getFaviconUrl = function(result) {
+  return 'chrome://universalsearch-skin/content/wikipedia-dark.svg';
+};
+WikipediaResult.prototype.getImageUrl = function(result) {
+  return result._get('enhancements.wikipedia.image.url');
+};
+WikipediaResult.prototype.getImageHeight = function(result) {
+  const height = result._get('enhancements.wikipedia.image.height');
+  const width = result._get('enhancements.wikipedia.image.width');
+  if (width <= height) {
+    return height * IMAGE_DIMENSION / width;
+  }
+  return IMAGE_DIMENSION;
+};
+WikipediaResult.prototype.getImageWidth = function(result) {
+  const height = result._get('enhancements.wikipedia.image.height');
+  const width = result._get('enhancements.wikipedia.image.width');
+  if (height < width) {
+    return width * IMAGE_DIMENSION / height;
+  }
+  return IMAGE_DIMENSION;
+};
+WikipediaResult.prototype.getUrl = function(result) {
+  return result._get('enhancements.wikipedia.url');
+};
+WikipediaResult.prototype.getTitleText = function(result) {
+  let suffix = / \- Wikipedia, the free encyclopedia$/;
+  return result._get('enhancements.wikipedia.title').replace(suffix, '');
+};
+
+
+
+function createElement(doc, elementName, id) {
```

Comment:
Seems weird to have this function floating around here. Move it to the BaseResult prototype